aten/src/ATen/native/cuda/UnaryOpsKernel.cu

```diff
@@ -9,6 +9,26 @@
 
 namespace at { namespace native {
 
+void abs_kernel_cuda(TensorIterator& iter) {
+  switch (iter.dtype()) {
+    case ScalarType::Bool:
+      gpu_kernel(iter, []GPU_LAMBDA(bool a) { return a; });
+      break;
+    case ScalarType::Half:
+      gpu_kernel(iter, []GPU_LAMBDA(at::Half a) -> half {
+        at::Half nega = -a;
```

Comment:
This has incorrect behavior for -0.0. `abs(-0.0) = 0.0`, but this will return `-0.0`.

(Please also add a test case for this if there isn't one present)